src/map/quest.c

```diff
@@ -314,10 +317,14 @@ static void quest_update_objective(struct map_session_data *sd, int mob_id)
 		qi = quest->db(sd->quest_log[i].quest_id);
 
 		for (j = 0; j < qi->objectives_count; j++) {
-			if (qi->objectives[j].mob == mob_id && sd->quest_log[i].count[j] < qi->objectives[j].count) {
-				sd->quest_log[i].count[j]++;
-				sd->save_quest = true;
-				clif->quest_update_objective(sd, &sd->quest_log[i]);
+			if ((qi->objectives[j].mob == 0 || qi->objectives[j].mob == md->class_) &&
+				sd->quest_log[i].count[j] < qi->objectives[j].count &&
+				(qi->objectives[j].level.min == 0 || qi->objectives[j].level.min <= md->level) &&
+				(qi->objectives[j].level.max == 0 || qi->objectives[j].level.max >= md->level) &&
```

Comment:
can you clarify something ?
the level range is set on the player side or the monster level ?

why check the monster level ?

Reply:
this is not meant to be used when a monster id is specified (i will add a warning for this later), the goal is to be able to have a quest which requires killing monsters of a certain level range, for example https://www.divine-pride.net/database/quest/16230